File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationPhotoViewHolder.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper;
import de.karzek.diettracker.presentation.model.GroceryDisplayModel;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeManipulationPhotoViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.recipe_image) ImageView imageView;

    private final OnDeleteImageClickListener onDeleteImageClickListener;

    public RecipeManipulationPhotoViewHolder(ViewGroup viewGroup,
                                             OnDeleteImageClickListener onDeleteImageClickListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_man_photo, viewGroup, false));
        ButterKnife.bind(this, itemView);

        this.onDeleteImageClickListener = onDeleteImageClickListener;
    }

    public void bind(RecipeManipulationViewItemWrapper item) {
        imageView.setImageBitmap(item.getImage());
    }

    @OnClick(R.id.delete_image) public void onDeleteImageClicked() {
        onDeleteImageClickListener.onDeleteImageClicked();
    }

    public interface OnDeleteImageClickListener {
        void onDeleteImageClicked();
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationPhotoViewHolder.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_man_photo.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeManipulationPhotoViewHolder(override val containerView: View,
                                        private val onDeleteImageClickListener: OnDeleteImageClickListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: RecipeManipulationViewItemWrapper) {
        recipe_image.setImageBitmap(item.image)
        delete_image.setOnClickListener { onDeleteImageClickListener.onDeleteImageClicked() }
    }

    interface OnDeleteImageClickListener {
        fun onDeleteImageClicked()
    }

}
